HomeTask/HomeTask/TaskList/TaskListViewModel.swift

```diff
@@ -44,44 +45,27 @@ class TaskListViewModel: ObservableObject {
         
     }
     
-    func validateNewTask(forTitle title: String) -> Bool {
-        return title.count > 1
-        
-    }
-    
-    func confirmPressed() {
-        let newTask = TaskModel(title: titleNewTask)
-        
-        if validateNewTask(forTitle: newTask.title) {
-            self.tasks.append(newTask)
-            showingAddNewTaskView.toggle()
-            titleNewTask = ""
-            textPlaceholder = "Nome da atividade"
-            
-        } else {
-            textPlaceholder = "A atividade precisa ter um nome"
-            
+    func swipeAction(task: TaskItem) {
+        coreDataManager.deleteTask(forTask: task) {
         }
+        self.loadData()
```

Comment:
Sobre o que eu comentei anteriormente, de podermos já passar a nova lista de tasks, para não precisar dar o fetch novamente, aqui teríamos algo como:

```suggestion
    func swipeAction(task: TaskItem) {
        coreDataManager.delete(task: task) { [weak self] result in
            guard let self else { return }
            
            switch result {
            case .success(let newTasks):
                self.tasks = newTasks
            case .failure(let error):
                // faz algo com o erro (exibe para o usuário? pede pra ele tentar novamente?)
            }  
        }
```

Reply:
So esse exemplo ai que tu colocou ja demonstrou com força pq o uso de result seria importante. vou tentar implementar esse cara e qualquer coisa falo contigo se tiver dificuldades